CRM/Contribute/Import/Parser/Contribution.php

```diff
@@ -626,19 +615,14 @@ private function deprecatedFormatParams($params, &$values, $create = FALSE) {
 
       switch ($key) {
         case 'contact_id':
-          if (!CRM_Utils_Rule::integer($value)) {
-            return civicrm_api3_create_error("contact_id not valid: $value");
```

Comment:
@eileenmcnaughton why not just throw an exception here?

Reply:
Cos it has already happened earlier on - that is a redundant check that was done during `validate`